include/deal.II/numerics/data_postprocessor.h

```diff
@@ -38,15 +40,189 @@ DEAL_II_NAMESPACE_OPEN
+   *       {
+   *         const typename hp::DoFHandler<dim>::cell_iterator
+   *           current_cell = input_data.template get_cell<hp::DoFHandler<dim> >();
+   *         const viscosity = look_up_viscosity (current_cell->material_id());
+   *
+   *         for (unsigned int q=0; q<input_data.solution_gradients.size(); ++q)
+   *           computed_quantities[q][0] = (viscosity * input_data.solution_gradients[q]).norm();
+   *       }
+   *   };
+   * @endcode
+   *
+   * @author Wolfgang Bangerth, 2016
+   */
+  template <int spacedim>
+  struct CommonInputs
+  {
+    /**
+     * An array of vectors normal to the faces of cells, evaluated at the points
+     * at which we are generating graphical output. This array is only used by
+     * the DataOutFaces class, and is left empty by all other classes for
+     * which the DataPostprocessor framework can be used. In the case of
+     * DataOutFaces, the array contains the outward normal vectors to the
+     * face, seen from the interior of the cell.
+     *
+     * This array is only filled if a user-derived class overloads the
+     * DataPostprocessor::get_needed_update_flags(), and the function
+     * returns (possibly among other flags)
+     * UpdateFlags::update_normal_vectors.  Alternatively, a class
+     * derived from DataPostprocessorScalar or DataPostprocessorVector
+     * may pass this flag to the constructor of
+     * DataPostprocessorScalar or DataPostprocessorVector.
+     */
+    std::vector<Tensor<1, spacedim> > normals;
+
+    /**
+     * An array of coordinates corresponding to the locations at which
+     * we are generating graphical output on one cell.
+     *
+     * This array is only filled if a user-derived class overloads the
+     * DataPostprocessor::get_needed_update_flags(), and the function
+     * returns (possibly among other flags)
+     * UpdateFlags::update_quadrature_points.  Alternatively, a class
+     * derived from DataPostprocessorScalar or DataPostprocessorVector
+     * may pass this flag to the constructor of
+     * DataPostprocessorScalar or DataPostprocessorVector.
+     */
+    std::vector<Point<spacedim> >     evaluation_points;
+
+    /**
+     * Set the cell that is currently being used in evaluating the data
+     * for which the DataPostprocessor object is being called.
+     *
+     * This function is not usually called from user space, but is instead
+     * called by DataOut and similar classes when creating the object that
+     * is then passed to DataPostprocessor.
+     */
+    template <typename DoFHandlerType>
+    void
+    set_cell (const typename DoFHandlerType::cell_iterator &cell);
+
+    /**
+     * Query the cell on which we currently produce graphical output.
+     * See the documentation of the current class for an example on how
+     * to use this function.
+     */
+    template <typename DoFHandlerType>
+    typename DoFHandlerType::cell_iterator
```

Comment:
Good question. Most of the time the cell is indeed active, but not always -- if you overload `first_cell()` and `next_cell()`, you can also just output data on coarser levels. 

My reasoning for doing it this way was essentially this: I can allow to store anything in the `boost::any` (well, doh, given the name :-) ) but the more I allow to store, the more difficult it will be to get it back out because you have to specify more. For example, if I allow storing active cell iterators, then indeed the template argument of the `get_cell` function must not just be a `DoFHandlerType`, but indeed an `IteratorType`. I'm pretty sure people will then try to get back out an active cell iterator when only a cell iterator was put in, or the other way around. In other words, it gets complicated -- especially since the corresponding call to `set_cell` that *determines the type you can get out* happens inside the library, not in user code.

So I thought I'd make everyone's lives easier: the library sets `cell_iterator` objects, and consequently one can only get `cell_iterator` objects out. If you can only get `cell_iterator` objects out, I may as well just allow specifying the enclosing class.